src/arr/base/contracts.arr

```diff
@@ -33,13 +40,17 @@ data FieldFailure:
   | missing-field(loc, field) with:
     render-reason(self, loc, from-fail-arg):
       [ED.error:
-        [ED.para: ED.text("Missing field"), ED.code(ED.text(self.field)),
-          ED.text("is required at"), draw-and-highlight(self.loc)]]
+        [ED.para: ED.text("Missing field "), ED.code(ED.text(self.field)),
```

Comment:
The alternative, I think, is worse: we've had _so_ many bug reports about "missing space in error message X", and it's a pain to fix them one at a time.  I'd rather have the common case be "we'll stick space in for you".  Another possibility is to eliminate `ED.para-nospace`, and instead create a `ED.nospace` constant that suppresses auto-inserted space in an `ED.para`.  Again, the common case is spaces are needed.  As it stands, you can't write the helper function the way you wish, since it sounds like you want a splicing context (i.e., have a helper function splice a bunch of content into some outer content), and since `ED.para` doesn't take a list argument (it's its own constructor), that helper can't be written directly.  So in any case you'll need to know what you're splicing into, and so you'll know whether there's space around it or not.